clock.go

```diff
@@ -20,7 +20,26 @@ func Equals(c1 Clock, c2 Clock) bool {
 	return timeToSeconds(c1) == timeToSeconds(c2)
 }
 
+// Between returns the duration in seconds between c1 and c2.
+// Note that the result is negative if c1 is after c2.
+func Between(c1 Clock, c2 Clock) int {
```

Comment:
Is int the right data-type here? I think it would be better to return a duration as that has a bit more semantic meaning for users.